File: components/tooltip/android/CiataTooltip.kt
package org.ciata.ds.tooltip

import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.PlainTooltip
import androidx.compose.material3.Text
import androidx.compose.material3.TooltipBox
import androidx.compose.material3.TooltipDefaults
import androidx.compose.material3.rememberTooltipState
import androidx.compose.runtime.Composable

/** Implementação experimental do CMP-0017 Tooltip para Jetpack Compose. */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun CiataTooltip(
    text: String,
    content: @Composable () -> Unit,
) {
    require(text.isNotBlank()) { "text não pode ser vazio." }

    TooltipBox(
        positionProvider = TooltipDefaults.rememberPlainTooltipPositionProvider(),
        tooltip = { PlainTooltip { Text(text) } },
        state = rememberTooltipState(),
    ) {
        content()
    }
}

File: components/tooltip/ios/CiataTooltip.swift
import SwiftUI

/// Implementação experimental do CMP-0017 Tooltip para SwiftUI.
public struct CiataTooltip<Content: View>: View {
    private let text: String
    private let appliesAccessibilityHint: Bool
    private let content: Content

    public init(
        text: String,
        appliesAccessibilityHint: Bool = true,
        @ViewBuilder content: () -> Content
    ) {
        let trimmed = text.trimmingCharacters(in: .whitespacesAndNewlines)
        precondition(!trimmed.isEmpty, "text não pode ser vazio")
        self.text = trimmed
        self.appliesAccessibilityHint = appliesAccessibilityHint
        self.content = content()
    }

    @ViewBuilder
    public var body: some View {
        if appliesAccessibilityHint {
            content
                .accessibilityHint(text)
        } else {
            content
        }
    }
}
